src/pages/index.js

```diff
@@ -39,34 +39,90 @@ function createCard(cardData) {
 function renderCard(cardData) {
   const card = new Card(cardData, "#card-template", {
     handleImageClick: () => imagePopup.open(cardData),
-  });
+  }, handleDeleteCardSubmit, toggleLikeCard);
   return card.getView();
 }
 
-// HANDLERS
-function handleFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  profileEditPopup.close();
+function toggleLikeCard(card) {
+    if(!card._isLiked) {
+      api.addLikeCard(card._id)
+      .then(() => {
+        card.toggleLikeButton();
+      })
+    } else {
+      api.removeLikeCard(card._id)
+      .then(() => {
+        card.toggleLikeButton();
+      });
+      }
+}
+
+// ~~~~~~~~~~~~~~~ HANDLERS ~~~~~~~~~~~~~~ //
+// -------- PROFILE ------- //
+function handleProfileEditSubmit(data) {
+  profileEditPopup.isLoading(true, "Saving...");
+  api.editUserInfo(data)
+  .then((userData) => {
+    newProfileInfo.setUserInfo(userData);
+    profileEditPopup.close();
+  })
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => profileEditPopup.isLoading(false));
 }
 
+function handleProfileImageSubmit(data) {
+  profileImageEditPopup.isLoading(true, "Saving...");
+  api.editProfileImage(data.url)
+  .then((res) => {
+    newProfileInfo.setProfileImage(res.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => {
+    profileImageEditPopup.isLoading(false);
+    profileImageEditPopup.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.

Fix that here and for all the other cases